build/i2c/src/lib.rs

```diff
@@ -967,8 +971,8 @@ impl ConfigGenerator {
 {indent}    Controller::I2C{controller},
 {indent}    PortIndex({port}),
 {indent}    {segment},
-{indent}    {address:#x}
-{indent}){refdes_part}"##,
+{indent}    {address:#x},{component_id}
```

Comment:
```suggestion
{indent}    {address:#x},
{indent}    {component_id},
```
?

Reply:
If the component ID isn't present, the `component_id` string will be empty, so this would generate two `,`s in a row, resulting in a syntax error. I included the comma in `component_id` instead, so that we only emit it if the string is non-empty.

Similarly, including the indent and newline will generate that unconditionally, even when `component_id` is the empty string. That's not a syntax error but it looks kinda weird. Therefore, I also formatted the indent and newline into the string:

https://github.com/oxidecomputer/hubris/blob/ca76539b16570d4d3bfe91c5e76f136324cd4244/build/i2c/src/lib.rs#L955-L965

That results in _this_ format! macro looking kinda bad, but it's the way to get the best output IMO.